carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -404,23 +405,19 @@ CARMAWorldModel::toRoadwayObstacle(const cav_msgs::ExternalObject& object) const
   }
 
   lanelet::BasicPoint2d object_center(object.pose.pose.position.x, object.pose.pose.position.y);
-  lanelet::BasicPolygon2d object_polygon = geometry::objectToMapPolygon(object.pose.pose, object.size);
 
-  auto nearestLanelet = semantic_map_->laneletLayer.nearest(
-      object_center, 1)[0];  // Since the map contains lanelets there should always be at least 1 element
+  auto nearestLaneletBoost = getIntersectingLanelet(object);
 
-  // Check if the object is inside or intersecting this lanelet
-  // If no intersection then the object can be considered off the road and does not need to processed
-  if (!boost::geometry::intersects(nearestLanelet.polygon2d().basicPolygon(), object_polygon))
-  {
+  if (!nearestLaneletBoost)
     return boost::none;
-  }
+  
+  lanelet::Lanelet nearestLanelet = nearestLaneletBoost.get();
 
   cav_msgs::RoadwayObstacle obs;
   obs.object = object;
   obs.connected_vehicle_type.type =
       cav_msgs::ConnectedVehicleType::NOT_CONNECTED;  // TODO No clear way to determine automation state at this time
-  obs.lanelet_id = nearestLanelet.id();
+  obs.lanelet_id = lanelet::utils::getId();
```

Comment:
This should be nearestLanelet.id()
The obs.lanelet_id refers to the id of the lanelet the object is on, not an id for the object.